front/components/assistant/conversation/ActionValidationProvider.tsx

```diff
@@ -21,88 +23,272 @@ import {
 import { useNavigationLock } from "@app/components/assistant_builder/useNavigationLock";
 import { useValidateAction } from "@app/hooks/useValidateAction";
 import type { MCPValidationOutputType } from "@app/lib/actions/constants";
+import type { BlockedToolExecution } from "@app/lib/actions/mcp";
 import { getIcon } from "@app/lib/actions/mcp_icons";
 import { useBlockedActions } from "@app/lib/swr/blocked_actions";
+import { useCreatePersonalConnection } from "@app/lib/swr/mcp_servers";
 import type {
   ConversationWithoutContentType,
   LightAgentMessageType,
   LightWorkspaceType,
-  MCPActionValidationRequest,
+  OAuthProvider,
+  OAuthUseCase,
 } from "@app/types";
 import { asDisplayName } from "@app/types";
 
-type ValidationQueueItem = {
+interface AuthenticationPageProps {
+  authActions: BlockedToolExecution[];
+  connectionStates: Record<string, "connecting" | "connected" | "idle">;
+  setConnectionStates: React.Dispatch<
+    React.SetStateAction<Record<string, "connecting" | "connected" | "idle">>
+  >;
+  createPersonalConnection: (params: {
+    mcpServerId: string;
+    mcpServerDisplayName: string;
+    provider: OAuthProvider;
+    useCase: OAuthUseCase;
+    scope?: string;
+  }) => Promise<boolean>;
+  errorMessage: string | null;
+}
+
+function AuthenticationPage({
+  authActions,
+  connectionStates,
+  setConnectionStates,
+  createPersonalConnection,
+  errorMessage,
+}: AuthenticationPageProps) {
+  const handleConnect = useCallback(
+    async (blockedAction: BlockedToolExecution) => {
+      setConnectionStates((prev) => ({
+        ...prev,
+        [blockedAction.actionId]: "connecting",
+      }));
+      const success = await createPersonalConnection({
+        mcpServerId: blockedAction.metadata.mcpServerId || "",
+        mcpServerDisplayName: blockedAction.metadata.mcpServerDisplayName || "",
+        provider: blockedAction.authorizationInfo!.provider,
+        useCase: "connection",
```

Comment:
the `useCase` is incorrect, should be the same as in `MCPServerPersonalAuthenticationRequired`
```suggestion
        useCase: "personal_actions",
```